scripts/verify_graph.py

```diff
@@ -117,6 +116,218 @@ def calculate_minimum_distance(raw_data, from_scene, to_scene, visited_scenes, c
         return None
 
     return minimum_distance_for_next_scenes + 1
+               
+
+def check_graph_routes(graph_data, raw_data, warning_prefix):
+    all_good = True
+
+    for from_vertex_key in graph_data["vertices"]:
+        for to_vertex_key in graph_data["vertices"]:
+            if from_vertex_key == to_vertex_key:
+                continue
+
+            last_edges = []
+            for edge in graph_data["edges"]:
+                if edge["to"] == to_vertex_key:
+                    last_edges.append(edge)
+
+            if len(last_edges) == 0:
+                continue
+
+            from_vertex = graph_data["vertices"][from_vertex_key]
+            to_vertex = graph_data["vertices"][to_vertex_key]
+
+            start_scene = from_vertex["furthest_scene"]
+            additional_length = 0
+            for task_key in from_vertex["tasks"]:
+                post_scene = raw_data["tasks"][task_key]["post_scene"]
+                if post_scene is not None:
+                    start_scene = post_scene
+                    additional_length = 1
+            
+            raw_routes = calculate_shortest_routes(raw_data, start_scene, to_vertex["furthest_scene"], [from_vertex["furthest_scene"]])
+
+            for calculated_length, task_obstacles in raw_routes:
+                expected_length = calculated_length + additional_length
+
+                allowed_condition_vertices = find_allowed_condition_vertices(graph_data["vertices"], raw_routes, last_edges, task_obstacles)
+
+                if allowed_condition_vertices is None:
+                    continue
+
+                max_length = expected_length + 1
+                result = calculate_shortest_graph_route(graph_data, from_vertex_key, to_vertex_key, allowed_condition_vertices, [from_vertex_key], 0, max_length)
+
+                if result is None:
+                    all_good = False
+                    print(warning_prefix + "Could not find a route from", from_vertex_key, "to", to_vertex_key, "(with expected task obstacles", task_obstacles, "and allowed conditions", str(allowed_condition_vertices) + ") either because there wasn't a route or the graph one was longer than the path (raw path length was", str(expected_length) + ")")
```

Comment:
the graph one was longer?

Reply:
underline that it might be because of too many conditions (for instance from ant_queen to mirror)